plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/ReserveVmInterfaceVspCommand.java

```diff
@@ -36,14 +37,26 @@
     String _vmUuid;
     String _vmUserName;
     String _vmUserDomainName;
-
-    public ReserveVmInterfaceVspCommand(String nicUuid, String nicMacAddress, String networkUuid, boolean isL3Network, String vpcUuid, String networkDomainUuid,
-            String networksAccountUuid, boolean isDomainRouter, String domainRouterIp, String vmInstanceName, String vmUuid, String vmUserName, String vmUserDomainName) {
+    boolean _useStaticIp;
+    String _staticIp;
+    String _staticNatIpUuid;
+    String _staticNatIpAddress;
+    boolean _isStaticNatIpAllocated;
+    boolean _isOneToOneNat;
+    String _staticNatVlanUuid;
+    String _staticNatVlanGateway;
+    String _staticNatVlanNetmask;
+
+    public ReserveVmInterfaceVspCommand(String nicUuid, String nicMacAddress, String networkUuid, boolean isL3Network, boolean isSharedNetwork, String vpcUuid, String networkDomainUuid,
+            String networksAccountUuid, boolean isDomainRouter, String domainRouterIp, String vmInstanceName, String vmUuid, String vmUserName, String vmUserDomainName,
+            boolean useStaticIp, String staticIp, String staticNatIpUuid, String staticNatIpAddress, boolean isStaticNatIpAllocated, boolean isOneToOneNat, String staticNatVlanUuid,
+            String staticNatVlanGateway, String staticNatVlanNetmask) {
```

Comment:
The number of parameters is extremely large -- creating a brittle interface and difficult to comprehend client client. Please refactor to the Builder pattern.